### PR TITLE
[CatalogPromotion][API] Reapplying catalog promotions after update its channel

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         |  master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    Background:
+        Given the store operates on a channel named "Web-US" with hostname "web-us"
+        And the store operates on another channel named "Web-GB" with hostname "web-gb"
+        And the store has a "T-Shirt" configurable product
+        And this product has "PHP T-Shirt" variant priced at "$100.00" in "Web-US" channel
+        And "PHP T-Shirt" variant priced at "$100.00" in "Web-GB" channel
+        And this product is available in "Web-US" channel and "Web-GB" channel
+        And there is a catalog promotion "Winter sale" available in "Web-US" channel that reduces price by "30%" and applies on "PHP T-shirt" variant
+        And there is a another catalog promotion "Christmas sale" available in "Web-US" channel and "Web-GB" channel that reduces price by "50%" and applies on "PHP T-shirt" variant
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:15]:
```suggestion
        And there is another catalog promotion "Christmas sale" available in "Web-US" channel and "Web-GB" channel that reduces price by "50%" and applies on "PHP T-shirt" variant
```

Comment (arti0090) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:15]:
```suggestion
        And there is an another catalog promotion "Christmas sale" available in "Web-US" channel and "Web-GB" channel that reduces price by "50%" and applies on "PHP T-shirt" variant
```

File: src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php [hunk 2]
```diff
@@ -177,6 +177,36 @@ public function thereIsACatalogPromotionAvailableInChannelThatReducesPriceByAndA
         $this->entityManager->flush();
     }
 
+    /**
+     * @Given /^there is a another catalog promotion "([^"]*)" available in ("[^"]+" channel) and ("[^"]+" channel) that reduces price by ("[^"]+") and applies on ("[^"]+" variant)$/
```
Comment (arti0090) [inline src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php:181]:
```suggestion
     * @Given /^there is an another catalog promotion "([^"]*)" available in ("[^"]+" channel) and ("[^"]+" channel) that reduces price by ("[^"]+") and applies on ("[^"]+" variant)$/
```

File: src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php [hunk 3]
```diff
@@ -177,6 +177,36 @@ public function thereIsACatalogPromotionAvailableInChannelThatReducesPriceByAndA
...
+     */
+    public function thereIsAAnotherCatalogPromotionAvailableInChannelAndChannelThatReducesPriceByAndAppliesOnVariant(
```
Comment (arti0090) [inline src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php:183]:
```suggestion
    public function thereIsAnAnotherCatalogPromotionAvailableInChannelAndChannelThatReducesPriceByAndAppliesOnVariant(
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Adam! :tada:

**Review by arti0090 — APPROVED** [review]

**Pull request merged**